src/scaler/spreadscaler/mod.rs

```diff
@@ -394,10 +400,46 @@ fn compute_spread(spread_config: &SpreadScalerProperty) -> Vec<(Spread, usize)>
         // there was a case that I didn't imagine
         Ordering::Greater => {
             warn!("Requesting more actor instances than were specified");
-            spreads
+            computed_spreads
         }
-        Ordering::Equal => spreads,
+        Ordering::Equal => computed_spreads,
+    };
+
+    // case when spread_config.spread.is_empty() <=> requested_spreads.is_empty()
+    if computed_spreads.len() == 1 {
+        return computed_spreads;
+    }
+
+    // this is an invariant check; the predicate should never be true.
+    // this check implicitly ensures that the reordering logic below is safe with regards to indexing.
+    if computed_spreads.len() != requested_spreads.len() {
+        panic!("Computed spreads and requested spreads are not the same length. This should not happen");
```

Comment:
I would err on removing this check or setting this as a warning log, see below comment re: panic philosophy